File: src/main/java/dev/manideeplanka/bookerang/services/BookService.java
package dev.manideeplanka.bookerang.services;

import dev.manideeplanka.bookerang.models.AddBookReq;
import dev.manideeplanka.bookerang.models.AddBookResult;
import dev.manideeplanka.bookerang.models.CopyDto;
import dev.manideeplanka.bookerang.repositories.BookRepository;
import lombok.extern.slf4j.Slf4j;
import java.util.List;

@Slf4j
public class BookService {


    BookRepository bookRepository;


    public BookService(BookRepository bookRepository) {
        this.bookRepository = bookRepository;
    }

    public AddBookResult addBook(AddBookReq req, String username) {
        return bookRepository.addBook(req.author(), req.title(), username);
    }

    public List<CopyDto> myBooks(String username) {
        return bookRepository.myBooks(username);
    }
}

File: internal/services/book_service.go
package services

import (
	"bookerang/internal/models"
	"bookerang/internal/repository"
	"context"
)

type BookService struct {
	repo *repository.BookRepository
}

func NewBookService(repo *repository.BookRepository) *BookService {
	return &BookService{repo: repo}
}

func (s *BookService) AddBook(ctx context.Context, req models.AddBookRequest, username string) (models.AddBookResult, error) {
	return s.repo.AddBook(ctx, req.Author, req.Title, username)
}

func (s *BookService) MyBooks(ctx context.Context, username string) ([]models.CopyDTO, error) {
	return s.repo.MyBooks(ctx, username)
}

func (s *BookService) NearbyBooks(ctx context.Context, username string, radius int64) ([]models.NearbyBookDTO, error) {
	return s.repo.NearbyBooks(ctx, username, radius)
}
